servicex/cache.py

```diff
@@ -1,14 +1,53 @@
 import json
 from pathlib import Path
 from typing import Dict, List, Optional, Tuple, Any
+from contextlib import contextmanager
 
 from .utils import ServiceXException, _query_cache_hash, sanitize_filename
 
+_ignore_cache = False
+
+
+@contextmanager
+def ignore_cache():
+    '''This will cause all caches to be ignored while it is invoked:
+
+    ```
+    with ignore_cache():
+        ServiceXDataset(...).get_data...()
+    ```
+
+    If you want to do this globally, you can just use the `__enter__()` method.
+    This is probably the only way to do this accross cells in a notebook.
+
+    ```
+    i = ignore_cache()
+    i.__enter__()
+    ... Query code, jupyter notebook cells, etc. go here
+    i.__exit(None, None, None)
+    ```
+
+    Note:
+
+    - The only time the cache is checked is when the query is actually made, not when
+      the servicex dataset object is created!
+    - Calls to this can be safely nested.
```

Comment:
It's a nice touch, but what's the use case for nested calls to `ignore_cache`?

Reply:
In my last implementation, with LINQ, I had this, and at one point I was deep in the code and wanted to make sure to ignore the cache - for testing - but then left it in for some reason, then did the ignore at a upper level, and I lost about 30 minutes of forking around to figure out what had happened. So now, in general, when I do somethign like this, I always make it nest proof. With the tools python has, this makes it totally trivial, actually.